project/source/Scripts/Objects/SourceContainer.gd

```diff
@@ -49,22 +49,25 @@ func SetupVisual():
 		fillsprite.show()
 		if contents: fillsprite.modulate = contents.color
 
-func SetOverrideImage(new):
+func SetOverrideImage(new: Texture2D) -> void:
 	image = new
 	SetupVisual()
 
-func SetContainerType(new):
+func SetContainerType(new: ContainerType) -> void:
 	containerType = new
 	SetupVisual()
 
-func CheckContents(group):
+func CheckContents(group: StringName) -> bool:
 	return contents.is_in_group(group)
 
-func TakeContents(volume = -1):
+func TakeContents(volume: int = -1) -> Array:
+	# TODO: This funciton originially had "return null" as the last line,
+	# but this does not work with static typing, so I replaced it with
+	# returning the substance because it is already null
 	if substance == null:
-		return null
+		return [substance]
```

Comment:
I'm not fully sure, but I kind of feel like it would be more apt to just return an empty array here. From the caller's perspective, this function returns an array of substances that have been taken from the container. The exact *reason* for this seems unimportant to me, and it significantly simplifies the usage if the caller can assume they got a valid array of substances and don't have to manually filter out `null`s.